src/main/java/com/xebialabs/overthere/smb/telnet/SmbTelnetConnection.java

```diff
@@ -49,8 +50,8 @@
      */
     public SmbTelnetConnection(String type, ConnectionOptions options, AddressPortMapper mapper) {
```

Comment:
I think you missed the intention of my comment. I think we should have 1 SmbConnection, and we pass it the right ProcessConnection in the ConnectionBuilder. So we don't need 3 subclasses for the Telnet, WinRM and WinRS cases. Same goes for the CIFS Connection.